File: presentation/src/main/java/example/com/presentation/ui/main/MainContract.java
package example.com.presentation.ui.main;

import java.util.Collection;

import example.com.presentation.models.GithubRepositoryModel;
import example.com.presentation.ui.base.IView;

/**
 * Created by Mohammad Jafarzadeh Rezvan on 10/11/2018.
 */
public interface MainContract {

    interface View extends IView {
        void githubRepositoriesLoaded(Collection<GithubRepositoryModel> githubRepositoryModels);
        void githubRepositoriesFailed(Throwable t);
    }

}

File: presentation/src/main/java/example/com/presentation/ui/main/MainContract.kt
package example.com.presentation.ui.main

import example.com.presentation.models.GithubRepositoryModel
import example.com.presentation.ui.base.BaseContract

/**
 * Created by Mohammad Jafarzadeh Rezvan on 10/11/2018.
 */
interface MainContract {

    interface View : BaseContract.View {
        fun githubRepositoriesLoaded(githubRepositoryModels: List<GithubRepositoryModel>)
        fun githubRepositoriesFailed(t: Throwable)
    }

    interface Presenter<T: View>: BaseContract.Presenter<T>
}
